lib/utils/ConfigParser.js

```diff
@@ -157,29 +159,8 @@ class ConfigParser {
          * If spec is a file on disk, it is set as the current spec, if it is not, it
          * is treated as a string match filter for the multiple-spec specification.
          */
-        if (typeof object.spec === 'string') {
-            const specs = new Set()
-            const specList = object.spec.split(/,/g)
-            const specsList = ConfigParser.getFilePaths(this._config.specs)
-
-            for (let spec of specList) {
-                if (fs.existsSync(spec) && fs.lstatSync(spec).isFile()) {
-                    specs.add(path.resolve(process.cwd(), spec))
-                } else {
-                    specsList.forEach(file => {
-                        if (file.match(spec)) {
-                            specs.add(file)
-                        }
-                    })
-                }
-            }
-
-            if (specs.size === 0) {
-                throw new Error(`spec file ${object.spec} not found`)
-            }
-
-            this._config.specs = [...specs]
-        }
+        this.setFilePathToFilterOptions(object.spec)
+        this.setFilePathToFilterOptions(object.exclude, true)
```

Comment:
Can we change the structure of this fn to
```js
this._config.specs = this.setFilePathToFilterOptions(object.spec)
this._config. exclude = this.setFilePathToFilterOptions(object. exclude)
```
this way it makes more sense and is better to test.

Reply:
we have to move the typeof condition up to the merge implementation to go down this path... I will push it so you see how it looks... I have mixed feelings about it...

Reply:
what's your thoughts? if it looks ok I will go down a similar route for the other PR...